### PR TITLE
feat(meta): implement Device Resource Client

### PR DESCRIPTION
**Pull request by weichou1229**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
implement Device Resource Client according to the doc https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceresource_profile__profileName__resource__resourceName_

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: v2/clients/http/deviceprofile.go [hunk 1]
```diff
@@ -136,3 +146,13 @@ func (client *DeviceProfileClient) DeviceProfilesByManufacturerAndModel(ctx cont
...
 }
+
+// DeviceResourceByProfileNameAndResourceName queries the device resource by profileName and resourceName
+func (client *DeviceProfileClient) DeviceResourceByProfileNameAndResourceName(ctx context.Context, profileName string, resourceName string) (res responses.DeviceResourceResponse, edgexError errors.EdgeX) {
+	requestPath := path.Join(v2.ApiDeviceResourceRoute, v2.Profile, url.QueryEscape(profileName), v2.Resource, url.QueryEscape(resourceName))
+	err := utils.GetRequest(ctx, &res, client.baseUrl, requestPath, nil)
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```
Comment (lenny-intel) [inline v2/clients/http/deviceprofile.go:157]: Missing caching of response. Just need a simple cache added to DeviceProfileClient that  can be a map of map of DeviceResourceResponse indexed by Profile name and then Resource name. I previous WG meeting we agreed no notification required as resources can only be added.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/clients/http/deviceprofile.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile.go:168]: > I previous WG meeting we agreed no notification required as resources can only be added.

Got it. Added cache.

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

File: v2/clients/http/deviceprofile.go [hunk 2]
```diff
@@ -136,3 +151,38 @@ func (client *DeviceProfileClient) DeviceProfilesByManufacturerAndModel(ctx cont
...
+	defer client.mux.RUnlock()
+	res, exists = client.resourcesCache[key]
+	return
+}
+
+func (client *DeviceProfileClient) setResourceWithMapKey(res responses.DeviceResourceResponse, key string) {
+	client.mux.Lock()
+	defer client.mux.Unlock()
+	client.resourcesCache[key] = res
+}
```
Comment (lenny-intel) [inline v2/clients/http/deviceprofile.go:182]: Perfect!

**Pull request merged**